query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/ViewColumnFilter.java

```diff
@@ -26,7 +39,11 @@ public ViewColumnFilter(Operator operator, Set<String> values) {
 
   public static ViewColumnFilter from(Config filterConfig) {
     Operator operator = Operator.valueOf(filterConfig.getString("operator"));
-    return new ViewColumnFilter(operator, new HashSet<>(filterConfig.getStringList("values")));
+    if (operator == Operator.EQ) {
+      return new ViewColumnFilter(operator, Set.of(filterConfig.getString("value")));
+    } else {
+      return new ViewColumnFilter(operator, new HashSet<>(filterConfig.getStringList("values")));
```

Comment:
`Set.copyOf` to match 43, and keep things immutable

Reply:
Done.